tools/const_finder/lib/const_finder.dart

```diff
@@ -35,16 +31,15 @@ class _ConstVisitor extends RecursiveVisitor<void> {
   // A cache of previously evaluated classes.
   static Map<Class, bool> _classHeirarchyCache = <Class, bool>{};
   bool _matches(Class node) {
-    assert(node != null);
-    final bool result = _classHeirarchyCache[node];
+    final bool? result = _classHeirarchyCache[node];
     if (result != null) {
       return result;
     }
     final bool exactMatch = node.name == className
         && node.enclosingLibrary.importUri.toString() == classLibraryUri;
     _classHeirarchyCache[node] = exactMatch
         || node.supers.any((Supertype supertype) => _matches(supertype.classNode));
-    return _classHeirarchyCache[node];
+    return _classHeirarchyCache[node]!;
```

Comment:
Nit: consider using a local variable for the result instead of doing an extra map lookup:

```dart
final bool match = exactMatch
        || node.supers.any((Supertype supertype) => _matches(supertype.classNode));
_classHeirarchyCache[node] = match;
return match;
```

Reply:
Done, although shouldn't the O(1) lookup be .. O(1)? :)